org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageClientImpl.java

```diff
@@ -115,7 +115,9 @@ public final CompletableFuture<ApplyWorkspaceEditResponse> applyEdit(ApplyWorksp
 			final var job = new Job(Messages.serverEdit) {
 				@Override
 				public IStatus run(IProgressMonitor monitor) {
-					LSPEclipseUtils.applyWorkspaceEdit(params.getEdit(), params.getLabel());
+					UI.getDisplay().syncExec(() -> {
+						LSPEclipseUtils.applyWorkspaceEdit(params.getEdit(), params.getLabel());
+					});
```

Comment:
Does this have to be part of this patch?

Reply:
Yeah, else you get InvalidThreadAccess when `workspace/applyEdit` is sent.

Reply:
The alternative would be to change `org.eclipse.lsp4e.LSPEclipseUtils.applyWorkspaceEditIfSingleOpenFile(WorkspaceEdit)`

Reply:
That said, it could be part of another PR/commit, but the bug we were running into is only fixed if it's done, too.

Reply:
I've reverted that change